sklearn/cluster/tests/test_hdbscan.py

```diff
@@ -143,13 +143,40 @@ def test_hdbscan_algorithms(algo, metric):
         hdb.fit(X)
 
 
-def test_hdbscan_dbscan_clustering():
+def test_dbscan_clustering():
     clusterer = HDBSCAN().fit(X)
     labels = clusterer.dbscan_clustering(0.3)
     n_clusters = len(set(labels) - OUTLIER_SET)
     assert n_clusters == n_clusters_true
 
 
+@pytest.mark.parametrize("outlier_type", _OUTLIER_ENCODING)
+@pytest.mark.parametrize("cut_distance", (0.1, 0.5, 1))
+def test_dbscan_clustering_outlier_data(outlier_type, cut_distance):
```

Comment:
This test does not check when both `np.inf` and `np.nan` appear in the dataset.

I am okay with a simple test with a dataset that contains `np.inf` and `np.nan` and not parameterize over `outlier_type`.